### PR TITLE
Map panel merg

### PR DESCRIPTION
**Pull request by wang-xiaotian**
Map GUI Panel

### CODE REVIEW

File: src/main/java/com/locallampoon/fiveh/core/Game.java [hunk 1]
```diff
@@ -17,11 +19,13 @@
     private static ArtPanel artPanel;
     private static ConsolePanel consolePanel;
     private static StatsPanel statsPanel;
+    private static MapPanel mapPanel;
 
     // CONSTRUCTOR
     public Game() {
         setHouseMap(XMLParser.parseRooms());
         this.player = new Player(houseMap.get("hall"));
+        GameMap.getInstance().setPlayer(this.player); // pass a reference to game map
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/core/Game.java]: Why do you need to `setPlayer`, if a reference is already initialized on line 27?

File: src/main/java/com/locallampoon/fiveh/data/rooms.xml [hunk 2]
```diff
@@ -118,6 +128,8 @@ glass bottles. Do those jars have eyeballs in them? The only door out is WEST to
             <item>Eyeball Jar</item>
         </roomItems>
         <npc>Bleacher Kid</npc>
+        <dx>71</dx>
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/data/rooms.xml:131]: I think this should be North of the Dining Room... Map displays it as East.

File: src/main/java/com/locallampoon/fiveh/core/Game.java [hunk 3]
```diff
@@ -89,6 +94,8 @@ private static void implementCommand(List<String> parsedCommandList, List<String
                 }
                 Room roomKeyID = houseMap.get(roomExits.get(dirMovement.getDirection()));
                 player.move(roomKeyID);
+                Game.mapPanel.getPanel().revalidate(); // highlight room
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/core/Game.java]: I think I understand that you should have revalidate() only on a move command, but is it possibly you'd ever take an action and end up moving somewhere without performing an explicit move command? Such as if you fell through a trap door or something?

File: src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        this.drawFloorLayout(graph);
+        boolean spotLight = false;
+        for(Map.Entry<String, Room> r : gameMap.getRooms().entrySet()){
+            if(gameMap.getPlayer().getCurrentRoom().equals(r.getValue())){
+                spotLight = true;
+            } else{
+                spotLight = false;
+            }
+            if(r.getKey().toLowerCase().equals("hall") || r.getKey().toLowerCase().equals("upstairslanding")){
+                this.drawHall(graph, r.getValue(), spotLight);
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java:45]: Your Hall looks like it is North of the Library, when in reality it is North-East.

File: src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            int y = y_center - MAP_ROOM_LENGTH * UNIT_SIZE/2 + i * UNIT_SIZE; // top left corner
+            graph.drawString("H", x_center- MAP_ROOM_LENGTH *UNIT_SIZE*2, y);
+            graph.drawString("H", x_center + MAP_ROOM_LENGTH *4 * UNIT_SIZE, y);
+        }
+        graph.drawString("H", x_center + MAP_ROOM_LENGTH *4 * UNIT_SIZE, y_center+ MAP_ROOM_LENGTH *UNIT_SIZE/2); // right bottom corner
+        // draw room name
+        graph.setColor(ROOM_TEXT_COLOR);
+        graph.setFont(ROOM_TEXT_FONT);
+        for(int i = 0; i < hall.getRoomName().length(); i++){
+            graph.drawString(String.valueOf(hall.getRoomName().charAt(i)),x_center- MAP_ROOM_LENGTH *UNIT_SIZE/2 + i * UNIT_SIZE,y_center- MAP_ROOM_LENGTH *UNIT_SIZE/2 - UNIT_SIZE);
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java:164]: Separate these lines with line breaks for readability.

File: src/main/java/com/locallampoon/fiveh/core/Game.java (hunk 1 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/core/Game.java]: I need the player's reference in the GameMapPanel to get its current position. Will generate a getPlayer() method inside Game or remove it if I can draw the map without using Player.getCurrentRoom()

File: src/main/java/com/locallampoon/fiveh/data/rooms.xml (hunk 2 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/data/rooms.xml:131]: The Kitchen is at the north of the Dining Room. The coordinates will be updated.

File: src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java (hunk 4 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java:45]: Will change the floor layout and makes the user easier to navigate

File: src/main/java/com/locallampoon/fiveh/core/Game.java (hunk 1 above)
Comment (chaynsaw) [reply src/main/java/com/locallampoon/fiveh/core/Game.java]: I can see why you need this, this is setting it on GameMap not Game. Whoops.

File: src/main/java/com/locallampoon/fiveh/core/Game.java (hunk 3 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/core/Game.java]: The start() inside Game class is never called, so I inserted a line updateMapGUI() inside handleCommand()

File: src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java (hunk 5 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/mappanel/GameMapPanel.java:164]: ok

File: src/main/java/com/locallampoon/fiveh/data/rooms.xml (hunk 2 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/data/rooms.xml:131]: The layout will be fixed in another PR

**Review by chaynsaw — APPROVED** [review]

**Pull request merged**